lib/stats/stat.js

```diff
@@ -0,0 +1,106 @@
+import assert from 'assert';
+import base from 'taskcluster-base';
+
+import * as series from './series';
+
+export default class Stat {
+  constructor(config) {
+    assert(config.influx, 'Must supply an influx configuration');
+    assert(config.workerId, 'Worker ID is required');
+    assert(config.workerType, 'Worker type is required');
+    assert(config.workerGroup, 'Worker group is required');
+    assert(config.workerNodeType, 'Worker instance type is required');
+    assert(config.provisionerId, 'Provisioner ID is required');
+
+    this.influx = new base.stats.Influx(config.influx);
+    this.baseWorkerStats = {
+      workerId: config.workerId,
+      workerGroup: config.workerGroup,
+      workerType: config.workerType,
+      instanceType: config.workerNodeType,
+      provisionerId: config.provisionerId,
+      capacity: config.capacity
+
+    };
```

Comment:
I definitely like the idea of adding tags when the series is created that will be added to all points recorded.  It makes a lot of sense .  Would simplify a bit in the provisioner as well where there are multiple things recorded with each series https://github.com/taskcluster/aws-provisioner/blob/master/provisioner/influx-series.js

Also, the purpose of the Stat class was not only to include certain columns with each series, but also not do things like:
https://github.com/taskcluster/aws-provisioner/blob/master/provisioner/aws-manager.js#L56-62

I figured calling stats.record(<series name>..) was simpler to read, but I might be mistaken.